commerce_coordinator/apps/commercetools/views.py

```diff
@@ -82,3 +98,56 @@ def post(self, request):
         send_order_confirmation_email(lms_user_id, customer.email, canvas_entry_properties)
 
         return Response(status=status.HTTP_200_OK)
+
+
+# noinspection DuplicatedCode
+class OrderSanctionedView(APIView):
+    """View to sanction an order and deactivate the lms user"""
+    permission_classes = [IsAdminUser]
+
+    def post(self, request):
+        """
+        Receive a message from commerce tools forwarded by aws event bridge
+        to sanction order and deactivate user through LMS
+        """
+        input_data = {
+            **request.data
+        }
+
+        logger.debug('[CT-OrderSanctionedView] Message received from commercetools with details: %s', input_data)
+
+        message_details = OrderMessageInputSerializer(data=input_data)
+        message_details.is_valid(raise_exception=True)
+
+        client = CommercetoolsAPIClient()
+        order_id = message_details.data['order_id']
+
+        try:
+            order = client.get_order_by_id(order_id)
+        except CommercetoolsError as _:  # pragma no cover
+            logger.error("Order not found: %s", order_id)
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        order_workflow_state = order.state.obj.key
+
+        customer = client.get_customer_by_id(order.customer_id)
```

Comment:
Suggestion:
- Please move `get_customer_by_id` into try/except block as well
- Change the logger function to exception as we need to be alerted on just exceptions
- Make exception message more meaningful and append API error as well

```
try:
    order = client.get_order_by_id(order_id)
    customer = client.get_customer_by_id(order.customer_id)
except CommercetoolsError as exc:
    logger.exception(f'Error fetching CT Customer or Order with ID "{order_id}". Error: {exc}')
    return Response(status=status.HTTP_404_NOT_FOUND)
```